src/commands/farm.rs

```diff
@@ -63,3 +101,27 @@ async fn prepare_farming() -> Result<FarmingArgs> {
         node,
     })
 }
+
+// TODO: have a callback here for updating the value of currently encoded
+fn _plotting_progress_bar(total_size: u64) {
+    let mut encoded = 0;
+    let to_be_plotted = total_size;
+
+    let pb = ProgressBar::new(total_size);
+    pb.set_style(
+        ProgressStyle::with_template(
+            "{spinner} [{elapsed_precise}] [{wide_bar}]
+     {bytes}/{total_bytes} ({bytes_per_sec}, {eta})",
+        )
+        .unwrap()
+        .progress_chars("#>-"),
+    );
+
+    while encoded < to_be_plotted {
+        encoded += 123123;
+        pb.set_position(encoded);
+        thread::sleep(Duration::from_millis(12));
+    }
+
+    pb.finish_with_message("downloaded");
+}
```

Comment:
WDUT about this dependency and using it:
https://docs.rs/zzz/latest/zzz/

Reply:
Hmm sounds interesting. Seamless integration with iterators is for sure nice to have. But the one I'm using is the most popular and allows more configuration (seems like). 
Don't have any strong opinions here. If you want me to switch to `zzz`, I can do that :)

Reply:
Oh, current library also has that feature:
https://github.com/console-rs/indicatif/blob/25afbed1ae264c95bf39bf88dc5b86941c65d543/examples/iterator.rs

Reply:
wonderful, I'll not resolve this convo to remind myself of updating this logic